extensions/microsoft-authentication/src/AADHelper.ts

```diff
@@ -872,6 +885,13 @@ export class AzureActiveDirectoryService {
 		// are already managing (see usages of `setSessionTimeout`).
 	}
 
+	private sessionMatchesEndpoint(session: IStoredSession): boolean {
+		// For older sessions with no endpoint set, it can be assumed to be the default endpoint
+		session.endpoint ||= defaultLoginEndpointUrl;
```

Comment:
look at you with the `||=` nice. Rarely see that one. Personally I prefer `??=` but this works.

Reply:
For strings I _usually_ use `||`, because it's rare that an empty string is different in any meaningful way from `null` / `undefined`.